documentation_quality_analysis/analyze_library/prioritize_api/llm/templates/one_shot_learning.txt

```diff
@@ -0,0 +1,7 @@
+{
+  "messages": [
+    {"role": "user",
+    "content": "I want to know if the core topic of a Stack Overflow post is a given API. I am giving two examples below. \nThis is a stack overflow post:\n‘The function numpy.savez() allows to store numpy objects in a file. Storing the same same object in two files results in two different files:\n\nimport numpy as np\nsome_array = np.arange(42)\nnp.savez('/tmp/file1', some_array=some_array)\nnp.savez('/tmp/file2', some_array=some_array)\nThe two files differ:\n\n$ diff /tmp/file1.npz /tmp/file2.npz \nBinary files /tmp/file1.npz and /tmp/file2.npz differ\nWhy aren't the files identical? Is there some random behavior, filename or time stamp included? Can this be workaround or fixed? (Is it a bug?)\n\nNote that this is not the case for np.save(). Files produced by np.save() are identical for identical inputs. So I guess it is related to zipping of the data.\n\nAFAICS there are only two bits different:\n\n$ xxd /tmp/file1.npz > /tmp/file1.hex\n$ xxd /tmp/file2.npz > /tmp/file2.hex\n$ diff -u0 /tmp/file1.hex /tmp/file2.hex    \n--- /tmp/file1.hex      2018-03-13 13:39:12.235897095 +0100\n+++ /tmp/file2.hex      2018-03-13 13:39:08.743927081 +0100\n@@ -1 +1 @@\n-0000000: 504b 0304 1400 0000 0000 ce6c 6d4c 9c9d  PK.........lmL..\n+0000000: 504b 0304 1400 0000 0000 cf6c 6d4c 9c9d  PK.........lmL..\n@@ -30 +30 @@\n-00001d0: 1403 1400 0000 0000 ce6c 6d4c 9c9d 6ad9  .........lmL..j.\n+00001d0: 1403 1400 0000 0000 cf6c 6d4c 9c9d 6ad9  .........lmL..j.\nI can't find any good hint in the implementation of the function, but I haven't checked the zipping code yet (also Python 3.6 might make a difference).\n\nNote: Tested with Python 2.7 and numpy 1.9.2.\n’\n\nIs the main topic of this post this API: 'numpy.save'? \n The correct answer is 'No'\n\nThis is another stackoverflow post: \n‘numpy.unique with order preserved:\n['b','b','b','a','a','c','c']\nnumpy.unique gives\n['a','b','c']\nHow can I get the original order preserved\n['b','a','c']\n’\nIs the main topic of this post this API: 'numpy.unique? \nThe correct answer is yes\n\n Now, this is another Stack Overflow post: ‘$post_body’\nIs the main topic of this post this API: '$api_fqn'? \nAnswer in yes or no\n"},
```

Comment:
I think the template looks quite confusing. I'd suggest that you have a json data file that has, for example, two to three examples along with their expected answer. 
The one shot learning template then has a placeholder for example1 while the few shot learning template has placeholders for 2-3 examples (whatever number you want). Then, in the code, you populate the templates with the examples read from that json data file. That way, it is easy to change the examples all in one place and the template becomes more readable.